examples/blog/src/layouts/BlogPost.astro

```diff
@@ -1,27 +1,107 @@
 ---
 import BaseHead from "../components/BaseHead.astro";
-import BlogHeader from "../components/BlogHeader.astro";
-import BlogPost from "../components/BlogPost.astro";
+import Header from "../components/Header.astro";
 
-const { content } = Astro.props;
-const { title, description, publishDate, author, heroImage, permalink, alt } = content;
+export interface Props {
+	content: {
+		title: string;
+		description: string;
+		publishDate: string;
+		heroImage?: {
+			src: string;
+			alt: string;
+		};
+	};
+}
+
+const {
+	content: { title, description, publishDate, heroImage },
+} = Astro.props as Props;
 ---
 
-<html lang={content.lang || "en"}>
+<html>
 	<head>
 		<link
 			rel="stylesheet"
 			href="https://cdn.jsdelivr.net/npm/prism-themes@1.9.0/themes/prism-lucario.css"
 		/>
-		<BaseHead {title} {description} {permalink} />
+		<BaseHead title={title} description={description} />
 	</head>
 
 	<body>
-		<BlogHeader />
-		<div class="wrapper">
-			<BlogPost {title} {author} {heroImage} {publishDate} {alt}>
-				<slot />
-			</BlogPost>
-		</div>
+		<Header />
+		<article class="wrapper content">
+			<header>
+				{heroImage && (
+					<img
+						width="720"
+						height="420"
+						class="hero-image"
+						loading="lazy"
+						src={heroImage.src}
+						alt={heroImage.alt}
+					/>
+				)}
+				<h1 class="title">{title}</h1>
+				<p class="publish-date">{publishDate}</p>
+			</header>
+			<main>
+				<slot></slot>
+			</main>
+		</article>
 	</body>
 </html>
+
+<style>
+	.hero-image {
```

Comment:
similar to above, we could simplify the selectors here

(now I'm realizing how annoying this nit might be!)